components/faction/commons/faction.lua

```diff
@@ -0,0 +1,126 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Faction
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Data = mw.loadData('Module:Faction/Data')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+local TypeUtil = require('Module:TypeUtil')
+
+local IconData = Lua.loadDataIfExists('Module:Faction/IconData')
+	or {byFaction = {}}
+
+local Faction = {propTypes = {}, types = {}}
+
+Faction.defaultFaction = Data.defaultFaction
+Faction.factions = Data.factions
+Faction.knownFactions = Data.knownFactions
+Faction.coreFactions = Data.coreFactions
+Faction.aliases = Data.aliases
+
+Faction.types.Faction = TypeUtil.literalUnion(unpack(Faction.factions))
+
+Faction.types.FactionProps = TypeUtil.struct{
+	bgClass = 'string?',
+	index = 'number',
+	name = 'string',
+	pageName = 'string?',
+	faction = Faction.types.Faction,
+}
+
+local byName = Table.map(Data.factionProps, function(faction, props) return props.name, faction end)
+local byLowerName = Table.map(byName, function(name, faction) return name:lower(), faction end)
+
+--- Checks if a entered faction is valid
+---@param faction string
+---@return boolean
+function Faction.isValid(faction)
+	return Data.factionProps[faction] ~= nil
+end
+
+--- Fetches the properties of an entered faction
+---@param faction string
+---@return table?
+function Faction.getProps(faction)
+	return Data.factionProps[faction]
+end
+
+--- Parses a faction from input. Returns the factions short handle/identifier.
+-- Returns nil if not a valid faction.
+---@param faction string
+---@return string?
+function Faction.read(faction)
+	if type(faction) ~= 'string' then
+		return nil
+	end
+
+	faction = faction:lower()
+	return Data.factionProps[faction] and faction
+		or byLowerName[faction]
+		or Faction.aliases[faction]
+end
+
+--- Returns the name of an entered faction identifier
+---@param faction string
+---@return string?
+function Faction.toName(faction)
```

Comment:
Should this include resolving shorthands/aliases by using `Faction.read`?
Or is it expected that the caller does this before?